src/api/Interpret.cc

```diff
@@ -1302,9 +1303,11 @@ void Interpret::getInterpolants(const ASTNode& n)
     vec<PTRef> grouping; // Consists of PTRefs that we want to group
     LetRecords letRecords;
     letRecords.pushFrame();
-    termNames.forEachNamedTerm([&](auto const & name, PTRef term) {
+    auto & termNames = main_solver->getTermNames();
```

Comment:
Also, why not keep the `forEachNamedTerm` method?
It would operate directly over the map.
This is a weird pattern, because you obtain all the keys and then you iterate over the map by doing a lookup for each key. Looks unnecessarily inefficient.

Reply:
Well, `forEachNamedTerm` always enforces to lookup the term for a name: `action(name, termByName(name));`, which is inefficient when we are only interested in names.
My code is actually equivalent to the previous version. I did not realize that it is inefficient.
Hence it would be better to iterate directly over the map. I will first change the iterators to the map iterators, then I will think of other options.
Using iterators is more transparent and compatible with STL algorithms. Also `forEachNamedTerm` does not allow `TAction` to have more arguments.